ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2950,21 +2957,32 @@ public Partition createPartition(Table tbl, Map<String, String> partSpec) throws
     int size = addPartitionDesc.getPartitionCount();
     List<org.apache.hadoop.hive.metastore.api.Partition> in =
         new ArrayList<org.apache.hadoop.hive.metastore.api.Partition>(size);
-    AcidUtils.TableSnapshot tableSnapshot = AcidUtils.getTableSnapshot(conf, tbl, true);
     long writeId;
     String validWriteIdList;
-    if (tableSnapshot != null && tableSnapshot.getWriteId() > 0) {
-      writeId = tableSnapshot.getWriteId();
-      validWriteIdList = tableSnapshot.getValidWriteIdList();
+
+    // In case of replication, get the writeId from the source and use valid write Id list
+    // for replication.
+    if (addPartitionDesc.getReplicationSpec().isInReplicationScope() &&
+        addPartitionDesc.getPartition(0).getWriteId() > 0) {
+      writeId = addPartitionDesc.getPartition(0).getWriteId();
+      validWriteIdList = new ValidReaderWriteIdList(TableName.getDbTable(tbl.getDbName(),
```

Comment:
Pls add a comment on why this hardcoding logic works for repl flow.

Reply:
Done.